plugins/CoreHome/ChangesHelper.php

```diff
@@ -0,0 +1,142 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Plugins\CoreHome;
+
+use Piwik\Common;
+use Piwik\Date;
+use Piwik\Db;
+use Piwik\Piwik;
+use Piwik\Plugin\Manager;
+use Piwik\Plugins\UsersManager\Model AS UsersModel;
+
+/**
+ * CoreHome changes helper class
+ */
+class ChangesHelper
+{
+    const NO_CHANGES_EXIST = 0;
+    const CHANGES_EXIST = 1;
+    const NEW_CHANGES_EXIST = 2;
+
+    /**
+     * Find any new version changes for a plugin
+     *
+     * @param $pluginName
+     * @param $version
+     */
+    public static function loadChangesForPlugin($pluginName, $version)
+    {
+        $file = Manager::getPluginDirectory($pluginName).'/changes.json';
+        if (file_exists($file)) {
+            $changes = file_get_contents($file);
+            if ($changes) {
+                $json = json_decode($changes, true);
+                if ($json && is_array($json)) {
+                    $json = array_reverse($json);
+                    foreach ($json as $change) {
+                        if (isset($change['version']) == $version) {
```

Comment:
Good spot. I've removed the version check, so an attempt will be made to insert all changes loaded from the plugin into the table, but since there is a unique index on plugin_name, version and title and `INSERT IGNORE` is being used then this will prevent duplicates. I'm trying to avoid getting into comparing plugin version numbers which could be in any format.